parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -431,6 +431,8 @@ ensureAddons() {
   rm -Rf ${ADDONS_DIR}/init
   replaceAddonsInit
   {{/* Force re-load all addons because we have changed the source location for addon specs */}}
+  {{/* Wait for the kube-addon manager to be running see pr 4743  */}}
+  ${KUBECTL} wait --for=condition=Ready --timeout=5m -l app=kube-addon-manager po
```

Comment:
Are we able to put a `|| exit_cse {{GetCSEErrorCode "ERR_ADDONS_START_FAIL"}}` condition in here? I'm afraid that if the pods don't come online in 5 mins that we'll be back in the same edge case outcome.

I've verified that that command returns a 1 exit code if the timeout is reached. Against a cluster that doesn't have a matching pod:

```
$ k wait --for=condition=Ready --timeout=1s -l app=kube-addon-manager po
error: no matching resources found
$ echo $?
1
```

Reply:
updated

Reply:
adding this seems to have show this doesn't work

Reply:
indeed, I'll build a cluster now and hop on during CSE (I'll change the timeout to a higher value to give more time to debug)